packages/types/src/message-metadata.ts

```diff
@@ -20,3 +26,287 @@ export interface MessageMetadata {
 export interface MessageMetadataEventPayloadObject {
   [key: string]: string | number | boolean;
 }
+
+// ------------------------------
+// Work object metadata
+// ------------------------------
+
+/**
+ * @description Application-specific entity and/or event data to attach to a Slack message.
+ * Provide 'entities' to set work object entity metadata.
+ * Provide 'event_type' and 'event_payload' to set event metadata.
+ */
+export interface EntityAndEventMessageMetadata extends Partial<MessageMetadata> {
```

Comment:
Since this seems to only be used to define `ChatPostMessageMetadata` in `packages/web-api/src/types/request/chat.ts`, what would you think about removing `EntityAndEventMessageMetadata` and instead defining it in `ChatPostMessageMetadata`, this would follow a similar pattern as `EntityMetadataUnfurls`

```ts
export interface ChatPostMessageMetadata {
  /**
   * @description Object representing message metadata, entity and/or event data to attach to a Slack message.
   * Provide 'entities' to set work object entity metadata.
   * Provide 'event_type' and 'event_payload' to set event metadata.
   */
  metadata?: Partial<MessageMetadata> & {
    /**
     * @description An array of work object entities.
     */
    entities?: EntityMetadata[];
  };
}
```